src/Database/LSMTree/Common.hs

```diff
@@ -243,4 +243,4 @@ listSnapshots (Session sesh) = Internal.listSnapshots sesh
 -- TODO: get rid of the @m@ parameter?
 type BlobRef :: (Type -> Type) -> Type -> Type
 type role BlobRef nominal nominal
-data BlobRef m blob = forall h. Typeable h => BlobRef (Internal.BlobRef (Internal.Run m h))
+data BlobRef m blob = forall h. Typeable h => BlobRef (Internal.BlobRef (Internal.Run m (Handle h)))
```

Comment:
Shouldn't this be

```haskell
BlobRef (Internal.BlobRef (Handle h)`
```

or is Run -> Handle happening later?

Reply:
Yes, not in this PR. In this PR I'm just following the existing strategy.

I plan to make a separate PR that changes the BlobRef representation to use a pair of a handle and a ref count, and hopefully that can be our unified representation for blob refs into both runs and the write buffer.